### PR TITLE
[tune] Add UUID back to trial names

### LINKED ISSUE #7517: [rllib] Trial name hash is always 0
### What is the problem?

This is only a minor cosmetic issue, but when running RLLib with tune, the checkpoints and parameters are stored in the `ray_result` directory with a different hash for each new trial to identify it. Since a few days, this hash is always 0, which makes identifying the trial difficult.

- Ray version 0.9.0.dev0 (latest wheel)
- TensorFlow version 2.1.0 (GPU)
- Python version 3.6.9
- Linux 5.3.0-28-generic 30~18.04.1-Ubuntu SMP x86_64 x86_64 x86_64 GNU/Linux

### Reproduction (REQUIRED)
```
rllib train --run=PPO --env=CartPole-v0
```
Output:
```
[...]
== Status ==
Memory usage on this node: 10.2/62.8 GiB
Using FIFO scheduling algorithm.
Resources requested: 3/32 CPUs, 0/1 GPUs, 0.0/32.52 GiB heap, 0.0/11.23 GiB objects
Result logdir: /home/jb2270/ray_results/default
Number of trials: 1 (1 RUNNING)
+-----------------------+----------+-------+
| Trial name            | status   | loc   |
|-----------------------+----------+-------|
| PPO_CartPole-v0_00000 | RUNNING  |       |
+-----------------------+----------+-------+

```

If we cannot run your script, we cannot fix your issue.

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://ray.readthedocs.io/en/latest/installation.html).

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This adds back the UUID component of trial names, while preserving the counter.

TLDR: `trial_00001` => `trial_fe1fa_00001`

## Related issue number

Closes https://github.com/ray-project/ray/issues/7517

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/tune/suggest/basic_variant.py [hunk 1]
```diff
@@ -92,7 +102,7 @@ def _generate_trials(self, num_samples, unresolved_spec, output_path=""):
             raise TuneError("Must specify `run` in {}".format(unresolved_spec))
         for _ in range(num_samples):
             for resolved_vars, spec in generate_variants(unresolved_spec):
-                trial_id = "%05d" % self._counter
+                trial_id = self._uuid_prefix + ("%05d" % self._counter)
```
Comment (richardliaw) [inline python/ray/tune/suggest/basic_variant.py:105]: I think this should be done in the `Trial.generate_id` codepath so that other trial-generating algorithms like Bayesian Optimization can also share this.

Comment (ericl) [reply python/ray/tune/suggest/basic_variant.py:105]: I think the tradeoff is that then the counter wouldn't reset for each group of experiments.

Note that generate id already generates a UUID, this is just fixing the basic variant gen to have the hash in addition to the counter.

Comment (richardliaw) [reply python/ray/tune/suggest/basic_variant.py:105]: ah got it

**Review by richardliaw — APPROVED** [review]
